pyomo/gdp/util.py

```diff
@@ -79,24 +84,62 @@ def clone_without_expression_components(expr, substitute=None):
 
 
 def target_list(x):
-    if isinstance(x, ComponentUID):
+    deprecation_msg = ("In future releases ComponentUID targets will no "
+                       "longer be supported. Specify targets as a "
+                       "Component or list of Components.")
+    if isinstance(x, _ComponentBase):
+        return [ x ]
+    elif isinstance(x, ComponentUID):
+        deprecation_warning(deprecation_msg)
+        # [ESJ 08/22/2019] We are going to have to pass this through and deal
+        # with it in the transformations because we can't get the component here
+        # since we don't have the instance
         return [ x ]
-    elif isinstance(x, (_ComponentBase, string_types)):
-        return [ ComponentUID(x) ]
     elif hasattr(x, '__iter__'):
         ans = []
         for i in x:
-            if isinstance(i, ComponentUID):
+            if isinstance(i, _ComponentBase):
+                ans.append(i)
+            elif isinstance(i, ComponentUID):
+                deprecation_warning(deprecation_msg)
+                # Same as above...
                 ans.append(i)
-            elif isinstance(i, (_ComponentBase, string_types)):
-                ans.append(ComponentUID(i))
             else:
                 raise ValueError(
-                    "Expected ComponentUID, Component, Component name, "
-                    "or list of these.\n\tReceived %s" % (type(i),))
+                    "Expected Component or list of Components."
+                    "\n\tRecieved %s" % (type(i),))
 
         return ans
     else:
         raise ValueError(
-            "Expected ComponentUID, Component, Component name, "
-            "or list of these.\n\tReceived %s" % (type(x),))
+            "Expected Component or list of Components."
+            "\n\tRecieved %s" % (type(x),))
+
+# [ESJ 07/09/2019 Should this be a more general utility function elsewhere?  I'm
+#  putting it here for now so that all the gdp transformations can use it
+def is_child_of(parent, child, knownBlocks=None):
```

Comment:
Might want to document what this function does.

Reply:
I think it would be fair to have a future PR that adds a docstring to this and lifts it into pyomo.common.